app/controllers/course/condition/achievements_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Course::Condition::AchievementsController < Course::ConditionsController
+  before_action :achievement_condition=, only: [:edit, :update, :destroy]
```

Comment:
Your same callback should be able to handle both. Cancan allows that.

And the naming of your setter method is wrong. Setter methods have this convention. In your case you are calling it for effect, not an operator overload.